opm/autodiff/FlowMain.hpp

```diff
@@ -252,9 +253,67 @@ namespace Opm
 #endif
         }
 
+        /// checks cartesian adjacency of global indices g1 and g2
+        bool cartesianAdjacent(const Grid& grid, size_t g1, size_t g2) {
```

Comment:
Non-template free functions defined in a header file must be declared `inline` to have weak (non-unique) linkage. I notice that this rule was also broken by the two free functions already declared in this file (will not fix now to avoid messing with this PR, feel free to fix it).